### PR TITLE
🔀 MERGE: `v1.6.2` release

### PR DESCRIPTION
**Pull request by sphuber**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,22 @@
 # Changelog
 
+## v1.6.2 - 2021-04-26
+
+### Bug fixes
```
Comment (chrisjsewell) [inline CHANGELOG.md:5]:
```suggestion
[full changelog](https://github.com/aiidateam/aiida-core/compare/v1.6.1...v1.6.2) | [GitHub contributors page for this release](https://github.com/aiidateam/aiida-core/graphs/contributors?from=2021-03-31&to=2021-04-26&type=c)

This patch release is intended to include final fixes/features for the `1.x` series, before the new file repository implementation is merged into the `develop` branch (see [#4345](https://github.com/aiidateam/aiida-core/pull/4345)), and work towards finalising a `2.x` release commences.

### Bug fixes
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,22 @@
...
+- Respect nested output namespaces in `Process.exposed_outputs` [[#4863]](https://github.com/aiidateam/aiida-core/pull/4863)
+- `NodeLinkManager` now properly regenerates original nested namespaces from the flat link labels stored in the database. This means one can now do `node.outputs.some.nested.output` instead of having to do `node.outputs.some__nested__output`. The same goes for `node.inputs` [[#4625]](https://github.com/aiidateam/aiida-core/pull/4625)
+- Fix `aiida.cmdline.utils.decorators.with_dbenv` always loading the database. Now it will only load the database if not already loaded, as intended [[#4865]](https://github.com/aiidateam/aiida-core/pull/4865)
+
+### Features
+- Add the `account` option to the `LsfScheduler` scheduler plugin [[#4832]](https://github.com/aiidateam/aiida-core/pull/4832)
+
+### Documentation
+- Update ssh proxycommand instructions [[#4839]](https://github.com/aiidateam/aiida-core/pull/4839)
+- Add the "How to extend workflows" section [[#4562]](https://github.com/aiidateam/aiida-core/pull/4562)
```
Comment (chrisjsewell) [inline CHANGELOG.md:16]: explain this a bit further (extend workflows how?) and provide a link to where it is in the documentation

**Review by chrisjsewell — CHANGES_REQUESTED** [review]
a few minor changes; as mentioned before my "push" for changelogs is to ensure they should be "human readable" and not just a list of commits

**Comment by ramirezfranciscof** [comment]
On hold guys, `verdi node delete` seems to have broken

File: CHANGELOG.md (hunk 2 above)
Comment (sphuber) [reply CHANGELOG.md:16]: @ramirezfranciscof you approved this PR. Can you provide this info? I think @mbercx is on holiday

File: CHANGELOG.md (hunk 1 above)
Comment (sphuber) [reply CHANGELOG.md:9]: Done

**Comment by sphuber** [comment]
Fixed the issue that put this on hold and the branch is rebased, so this is ready for review

File: CHANGELOG.md (hunk 2 above)
Comment (ramirezfranciscof) [reply CHANGELOG.md]: Extend workflows refers to including calls to other workflows inside of your workflow and the tools to adjust the interface (mostly the expose inputs/outputs). It refers to [this section](https://aiida.readthedocs.io/projects/aiida-core/en/latest/howto/write_workflows.html#extending-workflows). That works or do you want a more "ready to use" description? (in which case I would need parameters on the format 😬 )

File: CHANGELOG.md (hunk 1 above)
Comment (ramirezfranciscof) [reply CHANGELOG.md:9]: Sorry to meddle, but I think saying these are the final fixes for `1.x` may transmit the wrong idea for some people (as we would still be doing any important bug fixes that came up). Perhaps we could write something more specific of why we released this and less "future oriented", like:

```suggestion
This patch release is mainly intended to include some important fixes for the nested namespaces of output links in workflows.
It also features a fix for the `gotocomputer` CLI command, an improvement in the commands that require the loading of the database, an additional option for the `LsfScheduler`, and some updates of the documentation
See the list below for more information.

These represent the latest work done in the code, being released now before merging in some major changes related to the file repository.

### Bug fixes
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,26 @@
...
+- `NodeLinkManager` now properly regenerates original nested namespaces from the flat link labels stored in the database. This means one can now do `node.outputs.some.nested.output` instead of having to do `node.outputs.some__nested__output`. The same goes for `node.inputs` [[#4625]](https://github.com/aiidateam/aiida-core/pull/4625)
+- Fix `aiida.cmdline.utils.decorators.with_dbenv` always loading the database. Now it will only load the database if not already loaded, as intended [[#4865]](https://github.com/aiidateam/aiida-core/pull/4865)
+
+### Features
+- Add the `account` option to the `LsfScheduler` scheduler plugin [[#4832]](https://github.com/aiidateam/aiida-core/pull/4832)
+
+### Documentation
+- Update ssh proxycommand instructions [[#4839]](https://github.com/aiidateam/aiida-core/pull/4839)
+- Add the ["How to extend workflows"](https://aiida-core.readthedocs.io/en/latest/howto/write_workflows.html#extending-workflows) section [[#4562]](https://github.com/aiidateam/aiida-core/pull/4562)
+- Add help in intro for when quicksetup fails [[#4838]](https://github.com/aiidateam/aiida-core/pull/4838)
```
Comment (ramirezfranciscof) [inline CHANGELOG.md:21]:
```suggestion
- Update ssh proxycommand section with instructions on how to handle cases where the SSH key needs to be specified for the proxy server [[#4839]](https://github.com/aiidateam/aiida-core/pull/4839)
- Add the ["How to extend workflows"](https://aiida-core.readthedocs.io/en/latest/howto/write_workflows.html#extending-workflows) section, explaining the use of the `expose_inputs` and `expose_outputs` features, as well as nested namespaces [[#4562]](https://github.com/aiidateam/aiida-core/pull/4562)
- Add help in intro for when quicksetup fails due to problems autodetecting the PostgreSQL settings [[#4838]](https://github.com/aiidateam/aiida-core/pull/4838)
```

File: CHANGELOG.md (hunk 2 above)
Comment (ramirezfranciscof) [reply CHANGELOG.md]: I added a comment on the relevant section, @chrisjsewell can you check it out to see if that is better?

File: CHANGELOG.md (hunk 1 above)
Comment (chrisjsewell) [reply CHANGELOG.md:9]: Nope disagree with the suggestion

File: CHANGELOG.md (hunk 3 above)
Comment (chrisjsewell) [reply CHANGELOG.md:21]: this looks good to me 👍

**Review by chrisjsewell — CHANGES_REQUESTED** [review]
yeh just update of the documentation points

**Comment by chrisjsewell** [comment]
and just to note, in case you get trigger happy lol, we have the release workflow now, so don't do it the manual way

**Comment by dev-zero** [comment]
@chrisjsewell about that workflow: is it possible that it doesn't create annotated tags?

I see all the tags in my clone, yet a `git describe` on `develop` falls back to v1.5.2 instead of the more recent v1.6.1 (despite v1.6.1 being an ancestor of current HEAD of `develop` due to the back-merge):

```
$ git describe
v1.5.2-151-gcf7a823e1
$ git describe v1.6.1
v1.5.2-133-ge4e1e513f
$ git describe --tags v1.6.1
v1.6.1
$ git describe --tags
v1.6.1
```

**Comment by chrisjsewell** [comment]
> is it possible that it doesn't create annotated tags?

no idea, why do you need annotated tags?

**Comment by dev-zero** [comment]
> > is it possible that it doesn't create annotated tags?
> 
> no idea, why do you need annotated tags?

I could now answer "to make `git describe`" work as intended (showing how far ahead a branch is from the last release), but there are more important reasons here: https://git-scm.com/book/en/v2/Git-Basics-Tagging

**Comment by sphuber** [comment]
> no it won't, why do you need annotated tags?

In addition, one could say that we have been doing it like that for a long time and changing it is fine, as long as there is a good reason for it. In general I am not a fan of just changing the way we do things "just because".

**Comment by chrisjsewell** [comment]
If we really need to, we can think about retroactively converting to an annotated tag, but we are not changing from the automated deployment protocol

**Comment by ramirezfranciscof** [comment]
Is it not possible to include the annotated tags in the workflow?

**Comment by sphuber** [comment]
I think the tag is still created manually? It is just when that is pushed that the automated deploy to PyPI workflow kicks in to action.

**Comment by chrisjsewell** [comment]
>  In general I am not a fan of just changing the way we do things "just because".

sorry, I'm putting my foot down here, it was not "just because"; a number of previous releases literally had to be re-done because of manual errors in the release process.
The point is that it will now only be deployed to PyPI if it passes the tests, and also the build+deployment process is standardised in a GH action container

**Comment by chrisjsewell** [comment]
But I'm sure we can work out how to make the tag annotated

**Comment by dev-zero** [comment]
Speaking about PyPI, it might be a good idea to regenerate the PyPI (and other) tokens at this point, due to https://about.codecov.io/security-update/ . And yeah, the codecov github action was pulling the same bash script without checksum checking until very recently: https://github.com/codecov/codecov-action/commit/040839f579c614b247366096f5d107b52bb5f4b2 .

**Comment by chrisjsewell** [comment]
> Speaking about PyPI, it might be a good idea to regenerate the PyPI (and other) tokens at this point,

thats already been done

**Comment by chrisjsewell** [comment]
> I think the tag is still created manually?

No that tag is not created manually, you just make a release and it does the rest: https://github.com/aiidateam/aiida-core/releases/new

**Comment by sphuber** [comment]
> sorry, I'm putting my foot down here, it was not "just because"; a number of previous releases literally had to be re-done because of manual errors in the release process.

My comment was not in reference of the automated deploy. I have never objected to this since you put that in place. This was directed to not making the tags annotated, which we have always done and was in the instructions on how to make a release. You changed this without discussion and apparently without argument. That is what I was referring to with "just because".

Also, it is not "a number" of releases that were incorrect. There has been just the one after I was no longer responsible for releases.

**Comment by greschd** [comment]
>  retroactively converting to an annotated tag

FYI, that might actually delete the release info from Github: https://stackoverflow.com/questions/5002555/can-a-lightweight-tag-be-converted-to-an-annotated-tag#comment110678542_5002886

**Comment by chrisjsewell** [comment]
Yeh I don't know, it's a bit of a pain that GitHub's own release mechanism doesn't already do this, or have the option to do it (as far as I'm aware)

**Comment by chrisjsewell** [comment]
anyhow @sphuber get a move on, lets get this sucker out 😜
the release workflow triggers on any semantic versioning tag, so this does not hold up the release as I can still push an annotated tag instead of using the github release to generate it

File: CHANGELOG.md (hunk 1 above)
Comment (giovannipizzi) [reply CHANGELOG.md:9]: I personally agree with @ramirezfranciscof - the changelog should not contain information on the intended future, but on what has been changed. We can still convey that this is supposed to be the last 1.x release in the [wiki page with the roadmap](https://github.com/aiidateam/aiida-core/wiki/AiiDA-release-roadmap) (which anyway needs to be updated), but I wouldn't write "in stone" in a changelog - e.g. if the work towards 2.0 takes months, there's a large probability that we might have to release 1.6.3 with a few important bug fixes. 

I think the sentence of Francisco is OK, but OK to rephrase it a bit or we might even remove it, I don't think it's critical for the changelog.

Comment (chrisjsewell) [reply CHANGELOG.md:9]: I still disagree, but it's not a hill worth dying on

Comment (sphuber) [reply CHANGELOG.md:9]: I also tend to agree with @giovannipizzi and @ramirezfranciscof and propose we remove any explicit mentions of timelines. I am OOO atm but we will update this shortly (also including the other suggestion by @ramirezfranciscof ) and then we can release it.

Comment (chrisjsewell) [reply CHANGELOG.md:9]: ok thats fine, but then please re-write it in a different way, because most of the added text is superfluous.

Comment (sphuber) [reply CHANGELOG.md:7]: I meant that I agreed with the sentiment that the timeline/roadmap shouldn't really be in the changelog. I agree that the replacement text is essentially superfluous. I don't really see why we would need any text to justify the patch. It is a patch release. It contains bug fixes. It merits to be released. The list is there to say _what_ changed.

**Review by ramirezfranciscof — APPROVED** [review]
Ready to go for me

**Review by chrisjsewell — APPROVED** [review]

**Pull request merged**